src/focus.service.ts

```diff
@@ -490,14 +351,42 @@ export class FocusService {
         this.selected.click();
         return true;
       }
-    } else if (ev.event === Direction.BACK) {
-      this.location.back();
-      return true;
     }
 
     return false;
   }
 
+  private getFocusableElement(direction: Direction, root: HTMLElement, refRect: ClientRect): HTMLElement | null {
+    let el: HTMLElement | null | undefined = this.findNextFocusable(direction, root, refRect);
+    if (!el) { return null; }
+
+    const directive = this.registry.find(el);
+    if (directive && directive.arcFocusInside) {
+      el = this.getFocusableElement(direction, el, refRect);
+    }
+    return el;
+  }
+
+  private findNextFocusable(direction: Direction, root: HTMLElement, refRect: ClientRect) {
+    const directive = this.selected ? this.registry.find(this.selected) : undefined;
+    let nextElem: HTMLElement | null = null;
+
+    if (directive) {
+      nextElem = this.focusByRegistry.findNextFocus(direction, directive);
+    }
+
+    if (!nextElem && this.enableRaycast) {
+      nextElem = this.findNextFocusByRaycast(direction);
```

Comment:
If you're splitting things, it'd be cool to move this to its own strategy file too